### ISSUE TITLE
Migrating between different `SERIAL` sizes on postgresql does not work

### ISSUE THREAD

**Issue by tomhoule** [prisma/prisma#5206]
Steps to reproduce. In a schema with native types:

1. Create a model with a `id Int @id @default(autoincrement()) @db.Integer`
2. Migrate the database
3. Change the field to `id Int @id @default(autoincrement()) @db.SmallInt`
4. `prisma migrate dev`

```

warn "nativeTypes" is a preview feature. There may be bugs and it's not recommended to use it in production environments.
✔ Name of migration … smallintify
Error: Database error: Error querying the database: db error: ERROR: type "smallserial" does not exist
   0: migration_core::api::ApplyMigrations
             at migration-engine/core/src/api.rs:72
```

**Comment by darvelo** [comment]
I have a similar issue, but when migrating an unchanged (or changed) scheme that has `Unsupported("geometry")` in a few tables:

```

Error: P3006

Migration `20210221021406_initial` failed to apply cleanly to a temporary database.
Error:
Database error: Error querying the database: db error: ERROR: type "geometry" does not exist
   0: sql_migration_connector::flavour::postgres::sql_schema_from_migration_history
             at migration-engine/connectors/sql-migration-connector/src/flavour/postgres.rs:270
   1: sql_migration_connector::sql_database_migration_inferrer::calculate_drift
             at migration-engine/connectors/sql-migration-connector/src/sql_database_migration_inferrer.rs:40
   2: migration_core::api::DevDiagnostic
             at migration-engine/core/src/api.rs:79
```

I'm using PostGIS. Generating an initial migration on a fresh db works, but it seems if there's an existing migration having this type, the `prisma migrate dev --preview-feature` command fails.

**Comment by pantharshit00** [comment]
This is still reproducible in 2.22

**Comment by pantharshit00** [comment]
User report of this issue: https://github.com/prisma/prisma/issues/8746#issuecomment-904542665

**Comment by CompeyDev** [comment]
Facing the same issue with MySQL after changing the value of my `@id` to autoincrement.
![image](https://user-images.githubusercontent.com/74418041/181875652-0295ec09-5873-4fe7-bf62-6274a2686868.png)

Prisma migrate fails as well:
![image](https://user-images.githubusercontent.com/74418041/181875664-f55a8cd4-29aa-4c40-814a-4a80d81594ad.png)

**Comment by tomhoule** [comment]
Hi @CompeyDev , this is a different issue — the database error says the user you log in on mysql as can't create databases. The solution to this is in general to create another empty mysql, and use the `shadowDatabaseUrl` property in your datasource. Here's a link to the relevant docs: https://www.prisma.io/docs/concepts/components/prisma-migrate/shadow-database

**Comment by CompeyDev** [comment]
Yeah, I'm actually hosting the database on PlanetScale, so I suppose that's the reason this is happening. This is one of many issues that occurs with Prisma x PlanetScale.

**Comment by CompeyDev** [comment]
Temporarily, I switched from Int to String, and it works properly though.

**Comment by tomhoule** [comment]
On PlanetScale we usually recommend `db push` (in that case, no shadow database needed), because PlanetScale already has its own solution to manager migrations with branches.

**Comment by CompeyDev** [comment]
Yeah, and even once that's done, I still face the issue:
![image](https://user-images.githubusercontent.com/74418041/182187142-da09bf17-ce49-494d-8c7d-3e7a5599596d.png)

**Comment by tomhoule** [comment]
It may have to do with vitess (what PlanetScale builds on) having trouble dealing with tables without primary key, since your migration redefines the primary key. I'll try to get in touch with them.

**Comment by CompeyDev** [comment]
Alright, fair enough. I'll not bother going into such a minute detail for the moment and if this crops up again, I'll create an issue and let you know.

**Comment by templela** [comment]
This is still an issue. Would love to see a solution 🫶

**Comment by nanodino** [comment]
Still an issue in the latest release (4.16.0)
